examples/nextjs/pages/products/[handle].js

```diff
@@ -1,10 +1,19 @@
 import React from 'react';
+import { useRouter } from 'next/router';
 
 import $nacelle from 'services/nacelle';
 import ProductCard from 'components/ProductCard';
 import * as styles from 'styles/pages.styles';
 
 const ProductDetail = ({ product }) => {
+  const router = useRouter();
+
+  // If the page is not yet generated, this will be displayed
+  // initially until getStaticProps() finishes running
+  if (router.isFallback) {
```

Comment:
What are some of the instances where we'd be waiting on this? In the event of incremental static generation, does it completely wipe this page out as it's regenerating?

Reply:
Good question! This is what their docs say about it ([this section, specifically](https://nextjs.org/docs/basic-features/data-fetching#when-is-fallback-true-useful)):

> When someone requests a page that’s not generated yet, the user will see the page with a loading indicator. Shortly after, getStaticProps finishes and the page will be rendered with the requested data. From now on, everyone who requests the same page will get the statically pre-rendered page.

So I think the fallback UI will only be presented to the user when the page has expired and is being rebuilt on-the-fly.

Reply:
Indeed - it'll be up to the merchant developer to show a skeleton screen / something that isn't a weird experience for the fallback UI.